gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobReadChannel.java

```diff
@@ -46,4 +45,27 @@
    */
   void chunkSize(int chunkSize);
 
+  /**
+   * Saves the read channel state.
+   *
+   * @return an object that contains the read channel state and can restore it afterwards. State
+   *     object must implement {@link java.io.Serializable}.
+   */
+  public State save();
```

Comment:
no need for the public modifier here (and bellow, as long as they are defined in an interface).